src/homework9/linkedlist/LinkedList.java

```diff
@@ -0,0 +1,191 @@
+package homework9.linkedlist;
+
+public class LinkedList<T> implements List<T> {
+    private Node<T> node;
+    private int size;
+    private Node<T> prev;
+    private Node<T> first;
+    private Node<T> last;
+
+    private class Node<T> {
+        private Node<T> prev;
+        private Node<T> next;
+        private T element;
+
+        public Node(Node<T> prev, T element, Node<T> next) {
+            this.prev = prev;
+            this.next = next;
+            this.element = element;
+        }
+
+        public Node<T> getPrev() {
+            return prev;
+        }
+
+        public void setPrev(Node<T> prev) {
+            this.prev = prev;
+        }
+
+        public Node<T> getNext() {
+            return next;
+        }
+
+        public void setNext(Node<T> next) {
+            this.next = next;
+        }
+
+        public T getElement() {
+            return element;
+        }
+
+        public void setElement(T element) {
+            this.element = element;
+        }
+    }
+
+    public LinkedList() {
+        this.size = 0;
+    }
+
+    @Override
+    public void add(T value) {
+        if (size == 0) {
+            node = new Node(null, value, null);
+            first = node;
+            last = node;
+        } else {
+            prev = node;
```

Comment:
try create implementation without prev and node, we should use only first and last elements

Reply:
How I can set prev node in this block?
} else {
prev = node;
node = new Node(prev, value, null);
...
}